### ISSUE TITLE
entk sample error: no attribute '_uid'

### ISSUE THREAD

**Issue by aydinsaribudak** [radical-cybertools/radical.entk#403]
I couldn't run radical entk sample "/radical.entk/examples/user_guide/get_started.py" on two.radical-project using virtual environment. (It was working a couple weeks ago). I got "ImportError: cannot import name _remove_dead_weakref" error.

Then I installed conda environment, ran the same script by showing local.localhost as the resource location. The error message is:
AttributeError: 'NoneType' object has no attribute '_uid'
The details are with verbose.log file. The session folder is attached as well. 

This issue looks relevant to the issue below:
https://github.com/radical-cybertools/radical.entk/issues/355

Thanks,
Aydin
[session_logs.zip](https://github.com/radical-cybertools/radical.entk/files/3992853/session_logs.zip)
[verbose.log](https://github.com/radical-cybertools/radical.entk/files/3992854/verbose.log)

**Comment by aydinsaribudak** [comment]
radical-stack information:
```
/home/aydins/anaconda3/envs/ve-entk/lib/python2.7/site-packages/radical/saga
/home/aydins/anaconda3/envs/ve-entk/lib/python2.7/site-packages/radical/pilot
/home/aydins/anaconda3/envs/ve-entk/lib/python2.7/site-packages/radical/utils
/home/aydins/anaconda3/envs/ve-entk/lib/python2.7/site-packages/radical/entk

  python               : 2.7.17
  pythonpath           :
  virtualenv           : ve-entk

  radical.entk         : 0.72.1
  radical.pilot        : 0.72.2
  radical.saga         : 0.70.0
  radical.utils        : 0.70.1
```

**Comment by aydinsaribudak** [comment]
Using python3.6, and entk and pilot devel branches, I and Hyungro successfully completed the run. Since the issue is not replicated, I am closing the ticket.
Thanks!

**Comment by aydinsaribudak** [comment]
Hi,
I reopen the ticket because sample code get_started.py has not run successfully on xsede.bridges even though the run was successful on localhost which was set as the resource. The log files are attached. It is again same issue as mentioned previously:
AttributeError: 'NoneType' object has no attribute '_uid'

[sessionlogs.zip](https://github.com/radical-cybertools/radical.entk/files/3999490/sessionlogs.zip)
[verbose.log](https://github.com/radical-cybertools/radical.entk/files/3999491/verbose.log)

```
  python               : 3.6.9
  pythonpath           :
  virtualenv           : /home/aydins/myenv

  radical.entk         : 0.90.6-v0.72.0-91-gb6618476@devel
  radical.pilot        : 0.90.8-v0.90.1-223-g2917c34b@devel
  radical.saga         : 0.90.6
  radical.utils        : 0.90.8
```

**Comment by lee212** [comment]
I see the authentication error to XSEDE resources from `verbose.log` file:
```
radical.saga.exceptions.AuthenticationFailed: prompted for unknown password (XSEDE Authentication
password:)
```

I have noticed that `_uid` AttributeError is not the main cause but an indicator that you do not have access to the resource that you are trying to connect. You need to verify which access schema you use, and you have access with the schema. For example, Bridges you may choose `gsissh` and myproxy should be enabled for a certain period to connect.

**Comment by aydinsaribudak** [comment]
@lee212 thank you for looking into this. 

When I enter `gsissh -p 2222 bridges.psc.xsede.org` , I can successfully get connected to Bridges without putting any password. Myproxy is enabled as suggested.

How can I see the details of the access schema that I am using ?
Aydin

**Comment by aydinsaribudak** [comment]
@lee212 
After removing the project name "TG-MCB090174" from the resource description, the issue is resolved. Thank you for the suggestion.
I am closing the issue.
Aydin

**Comment by mturilli** [comment]
@aydinsaribudak, @lee212: How did the `TG-MCB090174` (wrong?) allocation triggered a login error?

**Comment by andre-merzky** [comment]
I had the same question: the project ID is used for the batch submission, but not to connect to the resource - so what else changed?

**Comment by lee212** [comment]
There is a missing record in this ticket, after resolving the login error, we found out this, the incorrect project name for Bridges is causing a failed batch submission and producing the `_uid` error message in a user screen.

**Comment by andre-merzky** [comment]
Ah, thanks for clarifying - I missed this.

**Comment by aydinsaribudak** [comment]
Hi,
I have to reopen this issue because I got the same type of error. I am using the sample script `get_started.py`. No matter I include project ID into resource description or not, I got the same error. The log files are generated for:
```
    res_dict = {
        'resource': 'xsede.bridges',
        'walltime': 10,
        'cpus': 1,
        'project': 'TG-MCB090174',
        'access_schema': 'gsissh'
    }
```

```
(env-test) aydins@js-17-94:~/radical.entk/examples/user_guide$ radical-stack

  python               : 3.7.6
  pythonpath           :
  virtualenv           : /home/aydins/env-test

  radical.entk         : 1.0.2
  radical.pilot        : 1.3.0
  radical.saga         : 1.3.0
  radical.utils        : 1.3.1
```

[radical.log](https://github.com/radical-cybertools/radical.entk/files/4532218/radical.log)
[session_folder.zip](https://github.com/radical-cybertools/radical.entk/files/4532219/session_folder.zip)
[verbose.log](https://github.com/radical-cybertools/radical.entk/files/4532220/verbose.log)

**Comment by andre-merzky** [comment]
I think you get the UID error because the UID fix was merged into devel, but that seems not released, yet.  @lee212 : do you need help with releasing 1.3?

The underlying problem is that your ssh connection to Bridges timed out.  When trying to debug RCT issues, please do a `grep ERROR *.log` on all log files, and check the context of all matches - that will usually give some idea about what went wrong.  

Does ssh to Bridges work on command line for you?  If so, please try again, might have been a network glitch.  Let me know if it continues to fail.  From where are you running?

**Comment by lee212** [comment]
@andre-merzky , Thanks! no, I was about to add the PR with the new release once updated docs are included.

**Comment by aydinsaribudak** [comment]
@andre-merzky 
 I am running my script from jetstream images, and ssh to bridges work on the command line.

Some error messages from <sessionID>/radical.log:
`1587781292.519 : radical.saga         : 19066 : 139914944521984 : ERROR    : skip adaptor radical.saga.adaptors.redis.redis_advert: import failed (No module named 'redis')`

`1587781330.053 : radical.saga.pty     : 19066 : 139914944521984 : ERROR    : Could not detect shell prompt (timeout) (/home/aydins/env-test/lib/python3.7/site-packages/radical/saga/utils/pty_shell_factory.py +301 (_initialize_pty)  :  raise rse.NoSuccess("Could not detect shell prompt (timeout)"))`

```
1587781330.054 : radical.entk.resource_manager.0000 : 19066 : 139914944521984 : ERROR    : Resource request submission failed
Traceback (most recent call last):
```

**Comment by andre-merzky** [comment]
Still, ssh times out.  Did you run again, i.e., is the error reproducible?

> Could not detect shell prompt (timeout)

**Comment by aydinsaribudak** [comment]
Yes I have the same error . Entk is with the devel branch. Should I checkout to master branch?

```
subject  : /C=US/O=National Center for Supercomputing Applications/CN=Aydin Saribudak
issuer   : /C=US/O=National Center for Supercomputing Applications/OU=Certificate Authorities/CN=MyProxy CA 2013
identity : /C=US/O=National Center for Supercomputing Applications/CN=Aydin Saribudak
type     : end entity credential
strength : 2048 bits
path     : /tmp/x509up_u1000
timeleft : 10:07:24
```

I addition to `radical.log` errors given before, the following is from `pmgr.0000.log`
```
1587844897.450 : pmgr.0000            : 29573 : 140113736607488 : DEBUG    : state push: pilot.0000: FAILED
1587844897.450 : pmgr.0000            : 29573 : 140113736607488 : DEBUG    : update pilot.0000
1587844897.451 : pmgr.0000            : 29573 : 140113736607488 : DEBUG    : call <bound method ComputePilot._default_state_cb of ['pilot.0000', 'xsede.bridges', 'FAILED']>
1587844897.451 : pmgr.0000            : 29573 : 140113736607488 : DEBUG    : pilot.0000 calls cb <bound method ComputePilot._default_state_cb of ['pilot.0000', 'xsede.bridges', 'FAILED']>
1587844897.451 : pmgr.0000            : 29573 : 140113736607488 : INFO     : [Callback]: pilot pilot.0000 state: FAILED.
1587844897.452 : pmgr.0000            : 29573 : 140113736607488 : ERROR    : [Callback]: pilot 'pilot.0000' failed (exit)
1587844897.452 : pmgr.0000            : 29573 : 140113736607488 : ERROR    : listener died
```

**Comment by andre-merzky** [comment]
Wait, you show us an gsissh certificate, but the command line I see in the logs is ssh, not gsissh.  Are you sure you use the correct access schema for the pilot (should be gsissh)?  And if you want to use ssh, are you sure you tested passwordless access with ssh, not gsissh?

**Comment by aydinsaribudak** [comment]
I just checked my res_dict. It is with gsissh. I am not sure why the command lines report ssh instead of gsissh.

```
    res_dict = {
        'resource': 'xsede.bridges',
        'walltime': 10,
        'cpus': 1,
        'project': 'TG-MCB090174',
        'access_schema': 'gsissh'
    }
```

**Comment by andre-merzky** [comment]
That might be - but in the logs I see ssh being used, not gsissh.  That explains the problem at least, but not why the wrong schema is selected.  This is the resource config which was used for the sessions you attached above, correct?

**Comment by andre-merzky** [comment]
Can you please try to run this on your command line:
```
/usr/bin/env TERM=vt100 X509_USER_PROXY=/tmp/x509up_u1000 /usr/local/globus-6/bin/gsissh -t -o Port=2222 -o ControlMaster=auto -o ControlPath=/tmp/saga_ssh_aydins_%h_%p.ctrl -o TCPKeepAlive=no -o ServerAliveInterval=10 -o ServerAliveCountMax=20 -o ConnectTimeout=10 bridges.psc.xsede.org hostname
```
and report the result?

**Comment by aydinsaribudak** [comment]
Here it is:
```
(env-test) aydins@js-17-94:~/DrugWorkflows$ /usr/bin/env TERM=vt100 X509_USER_PROXY=/tmp/x509up_u1000 /usr/local/globus-6/bin/gsissh -t -o Port=2222 -o ControlMaster=auto -o ControlPath=/tmp/saga_ssh_aydins_%h_%p.ctrl -o TCPKeepAlive=no -o ServerAliveInterval=10 -o ServerAliveCountMax=20 -o ConnectTimeout=10 bridges.psc.xsede.org hostname
br018.pvt.bridges.psc.edu
Connection to bridges.psc.xsede.org closed.
(env-test) aydins@js-17-94:~/DrugWorkflows$

```

**Comment by andre-merzky** [comment]
Ok - then please do the following:
```sh
export RADICAL_LOG_LVL=DEBUG
export RADICAL_SAGA_PTY_SSH_TIMEOUT=60
export RADICAL_SAGA_PTY_LOG_LVL=DEBUG
export RADICAL_SAGA_PTY_LOG_TGT=pty.log
```
and run your example again, and then please attach a tarball with the created log files and the session directory.  Thanks!

**Comment by aydinsaribudak** [comment]
Here is the tarball.

[session.tar.gz](https://github.com/radical-cybertools/radical.entk/files/4550593/session.tar.gz)

**Comment by andre-merzky** [comment]
Ah, here we go: the pty log shows:
```
mkfifo: cannot create fifo ‘/home/aydins/.radical/saga/adaptors/shell_job//fifo.29209’: Disk quota exceeded\nPROMPT-0->\n)
```

So, your disk quota are exceeded on Bridges, and thus RP won't be able to create any files, and is thus thrown out immediately after the login.

**Comment by aydinsaribudak** [comment]
@andre-merzky thank you for finding out the reason for this issue. 
I have removed all old session folders from `$SCRATCH` folder. (It is ` /pylon5/mc3bggp/aydins` ). However, I am still getting the same error when I run get_started.py script.
We have recently did several installations for workflow-2 project. I think this is the reason why I am exceeding the limit. I am closing the issue now. 
If we will decide to remove those new folders from my bridges account and I will keep receiving the same issue, I will reopen and let you know about it.
Thank you.
